hubblestack/extmods/modules/nebula_osquery.py

```diff
@@ -592,3 +902,293 @@ def _dict_update(dest, upd, recursive_update=True, merge_lists=False):
+        hash_md5 = md5()
+        hash_md5.update(file_content.encode('ISO-8859-1'))
+        new_hash = hash_md5.hexdigest()
+
+        if not os.path.isfile(hashfile):
+            f = open(hashfile, "w")
+            f.write(new_hash)
+            return False
+        else:
+            f = open(hashfile, "r")
+            old_hash = f.read()
+            if old_hash != new_hash:
+                log.info('old hash is {0} and new hash is {1}'.format(old_hash, new_hash))
+                log.info('changes detected in flag file')
+                return True
+            else:
+                log.info('no changes detected in flag file')
+    except:
+        log.error("some error occured, unable to determine whether osqueryd need to be restarted, not restarting osqueryd")
+    return False
+
+
+def _osqueryd_running_status_windows(servicename):
+    '''
+    This function will check whether osqueryd is running in windows systems
+    '''
+    log.info("checking if osqueryd is already running or not")
+    osqueryd_running = False
+    cmd_status = "(Get-Service -Name " + servicename + ").Status"
+    osqueryd_status = __salt__['cmd.run'](cmd_status, shell='powershell')
+    if osqueryd_status == 'Running':
+        osqueryd_running = True
+        log.info('osqueryd already running')
+    else:
+        log.info('osqueryd not running')
+        osqueryd_running = False
+    
+    return osqueryd_running
+
+
+def _start_osqueryd(pidfile, 
+                    configfile, 
+                    flagfile, 
+                    logdir, 
+                    databasepath, 
+                    servicename):
+    '''
+    This function will start osqueryd
+    ''' 
+    log.info("osqueryd is not running, attempting to start osqueryd")
+    if salt.utils.platform.is_windows():
+        log.error("requesting service manager to start osqueryD")
+        cmd = ['net', 'start', servicename]
+    else:
+        cmd = ['/opt/osquery/hubble_osqueryd', '--pidfile={0}'.format(pidfile), '--logger_path={0}'.format(logdir),
+               '--config_path={0}'.format(configfile), '--flagfile={0}'.format(flagfile), 
+               '--database_path={0}'.format(databasepath), '--daemonize']
+    __salt__['cmd.run'](cmd, timeout=10000)
+    log.info("daemonized the osqueryd")
+
+
+def _restart_osqueryd(pidfile, 
+                      configfile, 
+                      flagfile, 
+                      logdir, 
+                      databasepath, 
+                      hashfile, 
+                      servicename):
+    '''
+    This function will restart osqueryd
+    ''' 
+    log.info("osqueryd needs to be restarted, restarting now")
+
+    open_file = open(flagfile, 'r')
+    file_content = open_file.read().lower().rstrip('\n\r ').strip('\n\r')
+    hash_md5 = md5()
+    hash_md5.update(file_content.encode('ISO-8859-1'))
+    new_hash = hash_md5.hexdigest()
+
+    f = open(hashfile, "w")
```

Comment:
Either use 'with' clause or explicitly close 'f' file handle

Reply:
ok, I'll do this.